tests/gui/components/drive-selector.spec.js

```diff
@@ -0,0 +1,224 @@
+          }
+        }
+      }));
+    };
+
+    const injectDependencies = () => {
+      beforeEach(angular.mock.inject(function(_$controller_, _$q_, _$rootScope_) {
+        $controller = _$controller_;
+        $q = _$q_;
+        $rootScope = _$rootScope_;
+      }));
+    };
+
+    // Note '$rootScope' won't be instantiated until inside an 'it',
+    // therefore only call this function there or in a deeper scope.
+    const makeDriveSelectorController = () => {
+      const scope = $rootScope.$new();
+
+      return $controller('DriveSelectorController', {
+        $scope: scope,
+
+        // Inject stub for $uibModalInstance; this is necessary because
+        // none is provided by UI Bootstrap.
+        $uibModalInstance: {
+          result: $q.resolve(),
+          opened: $q.resolve(),
+          closed: $q.resolve(),
+          rendered: $q.resolve(),
+          close: _.constant($q.resolve()),
+          dismiss: _.constant($q.reject())
+        }
+      });
+    };
+
+    describe('.getButtonStatus()', function() {
+
+      importSelectionStateModel();
+      injectDependencies();
+
+      describe('given there is no selected drive', function() {
+
+        it('should return "disabled"', function() {
+
+          const DriveSelectorController = makeDriveSelectorController();
+
+          currentDrive = undefined;
+
+          const result = DriveSelectorController.getButtonStatus();
+
+          m.chai.expect(result).to.equal('disabled');
+
+        });
+      });
+
+      describe('given there is a protected drive', function() {
+
+        it('should return "disabled"', function() {
+
+          const DriveSelectorController = makeDriveSelectorController();
+
+          currentDrive = _.merge(defaultDrive, {
+            protected: true
+          });
+
+          const result = DriveSelectorController.getButtonStatus();
+
+          m.chai.expect(result).to.equal('disabled');
+
+        });
+      });
+
+      describe('given there is a system drive', function() {
+
+        it('should return "danger"', function() {
+
+          const DriveSelectorController = makeDriveSelectorController();
+
+          currentDrive = _.merge(defaultDrive, {
+            protected: false,
+            system: true
+          });
+
+          const result = DriveSelectorController.getButtonStatus();
+
+          m.chai.expect(result).to.equal('danger');
+
+        });
+      });
+
+      describe('given there is a removable drive', function() {
+
+        it('should return "primary"', function() {
+
+          const DriveSelectorController = makeDriveSelectorController();
+
+          currentDrive = _.merge(defaultDrive, {
+            protected: false,
+            system: false
+          });
+
+          const result = DriveSelectorController.getButtonStatus();
+          m.chai.expect(result).to.equal('primary');
+
+        });
+      });
+    });
+
+    describe('.getFooterMessage()', function() {
```

Comment:
Now that this function and `hasFooterMessage` are not needed anymore, we can inline the `beforeEach` blocks in `getButtonStatus` instead of using the utility functions.